Lesson5/school/diary/views.py

```diff
@@ -0,0 +1,81 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse
+from django.db import models
+from .models import User
+
+
+user = User(name="", password="", language="", grade="")
+
+
+def login(request):
+    global user
+
+    if request.method == 'POST':
+        # Info local save from HTML-Form:
+        l_name = request.POST.get('username')
+        l_password = request.POST.get('password')
+
+        # Log-In Button Pressed:
+        if request.POST.get('action') == 'Login':
+            if check_user_in_db(l_name, l_password):
+                # User Present:
+                return redirect(show_profile)
+            else:
+                # User Absent:
+                return render(request, 'login.html', {'message': 'No such User.'})
+
+
+        # Register Button Pressed:
+        if request.POST.get('action') == 'Register':
+            # Check if user PRESENT in DB:
+            if check_user_in_db(l_name, l_password):
+                return render(request, 'login.html', {'message': 'Username already exists.'})
+            else:
+                # Create a NEW USER
+                user = User(name=l_name, password=l_password)
+                user.save()
+                return redirect(show_profile)
+
+    return render(request, 'login.html')
+
+
+def show_profile(request):
+    global user
+
+    if request.method == 'POST':
+        if request.POST.get('action') == 'Change User':
```

Comment:
Your redirection to login won't change user, but will create a new one, think about working with your global variable to check that user exists and on the login URL we will update it not create a new instance.